k8s/migration/pkg/utils/credutils.go

```diff
@@ -561,10 +560,16 @@ func GetAllVMs(ctx context.Context, scope *scope.VMwareCredsScope, datacenter st
 	}
 	// Pre-allocate vminfo slice with capacity of vms to avoid append allocations
 	vminfo := make([]vjailbreakv1alpha1.VMInfo, 0, len(vms))
+
+	// Create a semaphore to limit concurrent goroutines to 100
+	semaphore := make(chan struct{}, 100)
+
 	for i := range vms {
-		wg.Add(1)
+		// Acquire semaphore (blocks if 100 goroutines are already running)
+		semaphore <- struct{}{}
 		go func(i int) {
-			defer wg.Done()
+			// Release semaphore when done
+			defer func() { <-semaphore }()
```

Comment:
@OmkarDeshpande7 Please put back wg.Done() and wg.Add(1) because this go routine can be in process and calling close on semaphore would be incorrect

Reply:
rectified